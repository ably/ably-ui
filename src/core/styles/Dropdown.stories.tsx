import React from "react";

export default {
  title: "CSS/Dropdown",
};

const exampleOptionFields = (
  <>
    <option value="select" disabled>
      Select language
    </option>
    <option value="javascript">Javascript</option>
    <option value="ruby">Ruby</option>
    <option value="python">Python</option>
    <option value="java">Java</option>
  </>
);

export const Default = {
  render: () => (
    <select className="ui-dropdown" defaultValue="select">
      {exampleOptionFields}
    </select>
  ),
};

export const Small = {
  render: () => (
    <select className="ui-dropdown-small" defaultValue="select">
      {exampleOptionFields}
    </select>
  ),
};

export const Disabled = {
  render: () => (
    <select className="ui-dropdown" defaultValue="select" disabled>
      {exampleOptionFields}
    </select>
  ),
};

export const Synthetic = {
  render: () => (
    <div className="ui-dropdown max-w-256" tabIndex={0}>
      Select language
    </div>
  ),
};

export const DarkMode = {
  render: () => (
    <div className="bg-neutral-1300 p-24">
      <select className="ui-dropdown ui-dropdown-dark" defaultValue="select">
        {exampleOptionFields}
      </select>
    </div>
  ),
};
