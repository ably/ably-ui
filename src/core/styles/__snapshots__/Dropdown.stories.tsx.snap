// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CSS/Dropdown DarkMode smoke-test 1`] = `
<div class="bg-neutral-1300 p-24">
  <select class="ui-dropdown ui-dropdown-dark">
    <option value="select"
            disabled
            selected
    >
      Select language
    </option>
    <option value="javascript">
      Javascript
    </option>
    <option value="ruby">
      Ruby
    </option>
    <option value="python">
      Python
    </option>
    <option value="java">
      Java
    </option>
  </select>
</div>
`;

exports[`CSS/Dropdown Default smoke-test 1`] = `
<select class="ui-dropdown">
  <option value="select"
          disabled
          selected
  >
    Select language
  </option>
  <option value="javascript">
    Javascript
  </option>
  <option value="ruby">
    Ruby
  </option>
  <option value="python">
    Python
  </option>
  <option value="java">
    Java
  </option>
</select>
`;

exports[`CSS/Dropdown Disabled smoke-test 1`] = `
<select class="ui-dropdown"
        disabled
>
  <option value="select"
          disabled
          selected
  >
    Select language
  </option>
  <option value="javascript">
    Javascript
  </option>
  <option value="ruby">
    Ruby
  </option>
  <option value="python">
    Python
  </option>
  <option value="java">
    Java
  </option>
</select>
`;

exports[`CSS/Dropdown Small smoke-test 1`] = `
<select class="ui-dropdown-small">
  <option value="select"
          disabled
          selected
  >
    Select language
  </option>
  <option value="javascript">
    Javascript
  </option>
  <option value="ruby">
    Ruby
  </option>
  <option value="python">
    Python
  </option>
  <option value="java">
    Java
  </option>
</select>
`;

exports[`CSS/Dropdown Synthetic smoke-test 1`] = `
<div class="ui-dropdown max-w-256"
     tabindex="0"
>
  Select language
</div>
`;
